### DISCUSSION TITLE
Unable to resolve role of

### DISCUSSION THREAD

**Discussion by konigbach** [cycle/orm#451] [Q&A] [answered]
Hello.
I'm trying to make a simple insertion like this:
```
class PriceRepository
{
    public function __construct(
        private readonly EntityManager $entityManager
    ) {
    }

    public function persist(Price $price): void
    {
        $this->entityManager->persist($price);
        $this->entityManager->run();
    }
}
```
the entity:
```
use Cycle\Annotated\Annotation\Column;
use Cycle\Annotated\Annotation\Entity;

#[Entity]
class Price
{
    public function __construct(
        #[Column(type: 'uuid')]
        private readonly string $id,
    ){
    }
}
```
But I have this error message Unable to resolve role of `App\Application\Price`.
I am using Spiral as framework. What am I missing? Thank you! :)

**Comment by roxblnfk** [comment, marked as answer]
Hi.
Looks like you've defined that repository as the Repository for the Entity (but I don't see it in the provided code).
In this case you should follow this guide https://cycle-orm.dev/docs/advanced-persist-repository/current/en

**Reply by konigbach** [reply]
hey @roxblnfk! Thanks for the help! I was tinkering a bit with the entities and the conclusion is that something's 'wrong' with the ids. If I put 'id' as primary it works. I will go in that direction because I need to work with Uuid6 as ids. Thanks again for the helping hand!
